Modules/_sqlite/connection.c

```diff
@@ -1998,9 +2003,16 @@ pysqlite_connection_iterdump_impl(pysqlite_Connection *self)
         }
         return NULL;
     }
-
-    PyObject *retval = PyObject_CallOneArg(iterdump, (PyObject *)self);
+    PyObject *args[2] = {(PyObject *)self, filter};
+    PyObject *kwnames = Py_BuildValue("(s)", "filter");
+    if (!kwnames) {
+        Py_DECREF(iterdump);
+        return NULL;
+    }
+    Py_ssize_t nargs = PyVectorcall_NARGS(1);
+    PyObject *retval = PyObject_Vectorcall(iterdump, args, nargs, kwnames);
```

Comment:
The following adjustments should do the trick (sorry if I pointed you in the wrong direction in the vector call PEP!):

```suggestion
    PyObject *args[3] = {NULL, (PyObject *)self, filter};
    PyObject *kwnames = Py_BuildValue("(s)", "filter");
    if (!kwnames) {
        Py_DECREF(iterdump);
        return NULL;
    }
    Py_ssize_t nargsf = 1 | PY_VECTORCALL_ARGUMENTS_OFFSET;
    PyObject *retval = PyObject_Vectorcall(iterdump, args+1, nargsf, kwnames);
```

Reply:
See https://peps.python.org/pep-0590/#py-vectorcall-arguments-offset

Reply:
Changed.